src/NATS.Client.Core/Commands/CommandWriter.cs

```diff
@@ -322,19 +322,18 @@ private static async Task ReaderLoopAsync(ILogger<CommandWriter> logger, ISocket
 
                 var buffer = result.Buffer;
                 var consumed = buffer.Start;
-
-                buffer = buffer.Slice(examinedOffset);
-                var examined = buffer.Start;
+                var examined = buffer.GetPosition(examinedOffset);
 
                 try
                 {
                     if (!buffer.IsEmpty)
                     {
-                        var bufferLength = (int)buffer.Length;
+                        var readBuffer = buffer.Slice(examinedOffset);
+                        var bufferLength = (int)readBuffer.Length;
```

Comment:
I _think_ (but I'm not entirely sure) that an empty write may possibly cause the `readBuffer` to be empty here, so probably good to check for `!readBuffer.IsEmpty` just to be careful